nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,344 @@
+
+## Create a Custom Field
+
+Navigate to the custom fields page by clicking on **Extensibility -> Custom Fields** in the Nautobot menu. Click on **Add** to create a new custom field.
+
+!!! note
+    Custom fields are initialized on associated objects when a content type is added to the custom field, including when the custom field is created. The initial value will be set to the `default` value of the custom field.
+
+### Custom Field Attributes
+
+#### Label
+
+The label is the human readable label of the custom field that will be displayed on the associated object detail view.
+
+![Custom Field Labels](./images/custom-fields/custom_field_detail_label.png)
+
+#### Grouping
+
+The optional grouping field allows you to group custom fields into collapsible menus.
+
+![Custom Field Grouping](./images/custom-fields/custom_field_detail_grouped.png)
+
+#### Slug
+
+The slug is used to create the URL endpoint for the custom field and is also used as the key in the custom field dictionary. This is automatically created from the label if not supplied. The default value should be sufficient for most deployments.
+
+#### Type
+
+The type of data that the custom field will store. Valid choices are documented in the custom field [model documentation](../models/extras/customfield.md#creating-custom-fields).
+
+#### Weight
+
+Weight determines how custom fields are sorted in forms and object detail views. Higher-weight fields will be ordered lower on the page.
+
+#### Description
+
+The description of a custom field is shown as a mouseover tooltip in object detail views and as help text under form fields.
+
+![Custom Field Edit](./images/custom-fields/custom_field_edit.png)
+
+#### Required
+
+Check the required box if this field cannot be null on the associated objects.
+
+!!! warning
+    If an associated object does not have a valid value assigned to a required custom field, that field must be updated with a valid value before the object can be saved. Try to supply a valid default value when creating required custom fields. Since automatic provisioning is only performed when a custom field's content types change, if an existing custom field is changed from optional to required the associated objects will have to be updated manually.
+
+#### Default
+
+The default value for the custom field. This form field only accepts JSON data so if you want to set the field default to a string of `foo` you must supply the JSON string `"foo"`. Boolean field valid values are `true` and `false` (all lowercase). Date fields are strings in the format `YYYY-MM-DD`.
+
+#### Filter Logic
+
++/- 1.4.0
+    Custom field [extended filtering](../rest-api/filtering.md#lookup-expressions) introduced extended lookup expression filters for `exact` and `icontains`, duplicating the functionality of both the `Strict` and `Loose` settings.
```

Comment:
This should really be called out in the appropriate section of `customfield.md` as well.